lib/text/simple_text_displayer.js

```diff
@@ -121,16 +121,29 @@ shaka.text.SimpleTextDisplayer = class {
     // We don't want to modify the array or objects passed in, since we don't
     // technically own them.  So we build a new array and replace certain items
     // in it if they need to be flattened.
-    const flattenedCues = cues.map((cue) => {
-      if (cue.nestedCues.length) {
-        const flatCue = cue.clone();
-        flatCue.nestedCues = [];
-        flatCue.payload = flattenPayload(cue);
-        return flatCue;
-      } else {
-        return cue;
+    // We also don't want to flatten the text payloads starting at a container
+    // element; otherwise, for containers encapsulating multiple caption lines,
+    // the lines would merge into a single cue. This is undesirable when a
+    // subset of the captions are outside of the append time window. To fix
+    // this, we only call flattenPayload() starting at elements marked as
+    // isContainer = false.
+    let flattenedCues = [];
+    const getCuesToFlatten = (cues) => {
+      for (const cue of cues) {
+        if (cue.isContainer) {
+          flattenedCues =
+              flattenedCues.concat(getCuesToFlatten(cue.nestedCues));
+        } else {
+          // Flatten the payload.
+          const flatCue = cue.clone();
+          flatCue.nestedCues = [];
+          flatCue.payload = flattenPayload(cue);
+          flattenedCues = flattenedCues.concat(flatCue);
```

Comment:
I was surprised to learn that Array.concat can be used for non-arrays, to append elements.  For clarity when pushing a single value, I would suggest you use Array.push instead, and we save Array.concat for concatenating two arrays.

Reply:
Done. Agreed this would be more readable, thanks!